### ISSUE TITLE
thread-first/last indentation change in Emacs 28 breaking the linter

### ISSUE THREAD

**Issue by ikappaki** [clojure-emacs/cider#3204]
The linter complains about `thread-first` argument indentation in a recent commit:

```
cider-client.el:600:0 (indent)   !                                    (cider-sync-tooling-eval)
cider-client.el:601:0 (indent)   !                                    (nrepl-dict-get "value")
cider-client.el:602:0 (indent)   !                                    read))
Found 3 warnings in file `cider-client.el'
```

This is because the indentation method has changed for `thread-first` and `thread-last` in Emacs 28 with [this commit](https://git.savannah.gnu.org/cgit/emacs.git/commit/?id=2a736738095c313ccef07d074aac4c5467b750e0) from `(indent 1)` to `(indent 0)`, and as a result their arguments are indented under the first argument, rather than the two spaces after the function:

auto indentation, before `(indent 1)`:
```emacs-lisp
(thread-first "(seq (.split (System/getProperty \"java.class.path\") \":\"))"
  (cider-sync-tooling-eval)
  (nrepl-dict-get "value")
  read)
```
after, Emacs 28, `(indent 0)`:
```emacs-lisp
(thread-first "(seq (.split (System/getProperty \"java.class.path\") \":\"))"
              (cider-sync-tooling-eval)
              (nrepl-dict-get "value")
              read)
```

There's a setting in `.dir-locals.el` for the linter to check the correct thread- indentation is at `(ident 1)`:
https://github.com/clojure-emacs/cider/blob/e86f2f74f65eaa3aa2dd088ad2e0b1426a04c3e1/.dir-locals.el#L30

any new `thread-*` calls written in Emacs 28 though, are most likely going to be indented with `(indent 0)` since this is the default.

As such, we have our first break with a recent check in identified by the linter (they are now indented under the first argument rather than two spaces after the function name):
```
cider-client.el:600:0 (indent)   !                                    (cider-sync-tooling-eval)
cider-client.el:601:0 (indent)   !                                    (nrepl-dict-get "value")
cider-client.el:602:0 (indent)   !                                    read))
Found 3 warnings in file `cider-client.el'
```

Shall I go ahead and change the `.dir-locals.el` `elisp-lint-indent-spaces` for `thread-first` and `thread-last` to be indent 0 (Emacs 28 setting), as well change every `thread-first` and `thread-last` call in CIDER to conform to this new indentation format? I much prefer the Emacs 27 format, but I don't think there is much we can do going forward ...

## Expected behavior

Linter should report any errors when checking the indentation for `thread-first`, `thread-last`

## Actual behavior

Linter complains about indentation of recent `thread-first` call.

## Steps to reproduce the problem

run `eldev lint`

### CIDER version information

cider master

### Emacs version

Emacs 28

### Operating system

Ubuntu 20.04

**Comment by vemv** [comment]
Hey there!

Thanks as always. https://github.com/clojure-emacs/cider/pull/3203 opportunistically fixes this.

Given your analysis, I'm actually not sure of why that PR is green? I'd think that either an Emacs 27 or an Emacs 28 job would fail.

**Comment by ikappaki** [comment]
> Thanks as always. #3203 opportunistically fixes this.
> 
> Given your analysis, I'm actually not sure of why that PR is green? I'd think that either an Emacs 27 or an Emacs 28 job would fail.

ok, so the corresponding linter setting in `.dir-local.el` instructs the linter to check the cider files using the old indentation format (i.e. `(indent 1)`). The recent commit has a `thread-first` sexp with `(indent 0)` format and thus the linter noticed the arguments were not indented correctly. You have corrected the relevant indentation with #3203 to be `(indent 1)` and that fixes it, the linter is happy.

Why was the new `third-first` entry was indented with the `(indent 0)` method? Because Emacs 28 now defaults to that. So, the committer most probably used Emacs 28 to indent the sexp, and thus it was committed under the first arguments as such.

The next person who will commit a thread first or last call, will most probably commit it with `indent 0` format, because this is what Emacs 28 defaults to now, and thus we will get more errors like this. Somebody then has to come and fix it after the commit (assuming they don't run the linter locally), so we get into a perpetual commit and fix thread- indentation work.

So my suggestion is to change all `thread-*` in the codebase to conform to `(indent 0)` and change the linter setting in `.dir-locals.el` to `(indent 0)`. Any new `thread-*` calls in Cider intended with Emacs 28 (by having the user pressing the TAB or Alt-Q) will default to `(indent 0)` and, everyone, I think will be happy.

Sorry, If I repeated again what I said earlier, just wanted to make sure we are all on the same page.

Thanks

**Comment by bbatsov** [comment]
> So my suggestion is to change all thread-* in the codebase to conform to (indent 0) and change the linter setting in .dir-locals.el to (indent 0). Any new thread-* calls in Cider intended with Emacs 28 (by having the user pressing the TAB or Alt-Q) will default to (indent 0) and, everyone, I think will be happy.

Fine by me.
